src/iguana/algorithms/clas12/MomentumCorrection.cc

```diff
@@ -0,0 +1,406 @@
+#include "MomentumCorrection.h"
+#include "iguana/algorithms/TypeDefs.h"
+#include <cmath>
+
+namespace iguana::clas12 {
+
+  REGISTER_IGUANA_ALGORITHM(MomentumCorrection);
+
+  void MomentumCorrection::Start(hipo::banklist& banks) {
+    CacheBankIndex(banks, "REC::Particle", b_particle);
+    CacheBankIndex(banks, "RUN::config", b_config);
+  }
+
+
+  void MomentumCorrection::Run(hipo::banklist& banks) const {
+    auto& particleBank = GetBank(banks, b_particle, "REC::Particle");
+    auto& configBank   = GetBank(banks, b_config,   "RUN::config");
+    ShowBank(particleBank, Logger::Header("INPUT PARTICLES"));
+
+    auto torus = configBank.getFloat("torus", 0);
+
+    for(int row = 0; row < particleBank.getRows(); row++) {
+
+      //////////////////////////////////
+      //////////////////////////////////
+      int sector = 1; // FIXME: get the true sector
+      //////////////////////////////////
+      //////////////////////////////////
```

Comment:
Implement sector retrieval and use it in `clas12::MomentumCorrection`

```suggestion
      auto sector = GetSector(_____);
```

Reply:
Tracking in https://github.com/JeffersonLab/iguana/issues/113

Not planned for this PR.